### PR TITLE
Added the MaxConcurrentShardRequests in JSONData for Elasticsearch datasources

### LINKED ISSUE #195: Unable to set "Max concurrent Shard Requests" for Elasticsearch in grafana_data_source
Hi,

In the latest version of Grafana, Elasticsearch datasources can be defined with **Max concurrent Shard Requests** = 
(Optional, integer) Maximum number of concurrent shard requests that each sub-search request executes per node. Defaults to 5.

But we are not able to specify the  **Max concurrent Shard Requests**  via the terraform grafana_datasource.

**Terraform Version**

 v0.15.0

**Affected Resource(s)**

    ̀json_data`  of `grafana_data_source`

Please note that the grafana-api-golang-client doesn't support **Max concurrent Shard Requests**, so changes to that would be required as well. PR: https://github.com/grafana/grafana-api-golang-client/pull/24

### PR DESCRIPTION
**Pull request by aymericDD**
# What

Adds "MaxConcurrentShardRequests" field to JsonData in the data source management for Elasticsearch.

# Why

In the latest version of Grafana, Elasticsearch datasources can be defined with Max concurrent Shard Requests =
(Optional, integer) Maximum number of concurrent shard requests that each sub-search request executes per node. Defaults to 5.

But we are not able to specify the Max concurrent Shard Requests via the terraform grafana_datasource.

This pull-request adds support to set the Max concurrent Shard Requests while creating Elasticsearch datasources and depends on PR https://github.com/grafana/grafana-api-golang-client/pull/24.

Fixes and closes #195.

### CODE REVIEW

**Comment by aymericDD** [comment]
The test are not working because we need this PR to be merged https://github.com/grafana/grafana-api-golang-client/pull/24

File: grafana/resource_data_source.go [hunk 1]
```diff
@@ -430,36 +435,36 @@ func makeDataSource(d *schema.ResourceData) (*gapi.DataSource, error) {
...
-		CustomMetricsNamespaces: d.Get("json_data.0.custom_metrics_namespaces").(string),
-		DefaultProject:          d.Get("json_data.0.default_project").(string),
-		DefaultRegion:           d.Get("json_data.0.default_region").(string),
-		Encrypt:                 d.Get("json_data.0.encrypt").(string),
-		EsVersion:               int64(d.Get("json_data.0.es_version").(int)),
-		GraphiteVersion:         d.Get("json_data.0.graphite_version").(string),
-		HTTPMethod:              d.Get("json_data.0.http_method").(string),
-		Interval:                d.Get("json_data.0.interval").(string),
-		LogLevelField:           d.Get("json_data.0.log_level_field").(string),
-		LogMessageField:         d.Get("json_data.0.log_message_field").(string),
```
Comment (trotttrotttrott) [inline grafana/resource_data_source.go:448]: Looks like `LogMessageField` got accidentally removed.

**Review by trotttrotttrott — COMMENTED** [review]
Looks good for the most part! Seems like we inadvertently lost a field. Your API PR got merged so you can now commit the vendor updates to get tests passing.

```
go get github.com/grafana/grafana-api-golang-client
go mod vendor
go mod tidy
```

File: grafana/resource_data_source.go (hunk 1 above)
Comment (aymericDD) [reply grafana/resource_data_source.go:448]: Oups... Sorry for that :innocent:

**Comment by aymericDD** [comment]
Thanks for your review @trotttrotttrott :heart: 
I dont know why the tests are crashed. I have no issue in local and the test was crashing is not in the scope of my feature. :thinking:

**Review by trotttrotttrott — APPROVED** [review]
Thanks @aymericDD!

I saw that failure yesterday too on a documentation update. Not sure why that started happening. But I restarted your build and it's succeeding now 👍

**Pull request merged**